tests/MongoDB.Driver.Tests/Linq/MongoQueryableTests.cs

```diff
@@ -1185,6 +1294,21 @@ public void SelectMany_with_collection_selector_method_computed_scalar()
                 "{ $project: { __fld0: { $add: ['$C.E.F', '$G.E.F', '$G.E.H'] }, _id: 0 } }");
         }
 
+        [Fact]
+        public void SelectMany_with_collection_selector_method_computed_scalar_which_is_called_from_SelectMany()
+        {
+            var query = CreateQuery()
+                .SelectMany(g => g.G)
+                .SelectMany(s => s.S, (x, c) => (int?)(x.E.F + c.E.F + c.E.H));
```

Comment:
`(int?)` is required here just because of these fields `x.E.F`, 'c.E.F', 'c.E.H' have null values in contents and cannot be converted to `int`.